Even if not defining __GLIBC__, musl libc has program_invocation_short_name

--- a/dfcompat.c	2013-06-03 16:03:15.000000000 +0200
+++ b/dfcompat.c	2015-08-07 14:57:37.814738238 +0200
@@ -105,8 +105,6 @@
 
 #ifndef HAVE_GETPROGNAME
 
-#ifdef __GLIBC__
-
 #include <errno.h>
 
 const char *
@@ -115,8 +113,4 @@
 	return (program_invocation_short_name);
 }
 
-#else /* __GLIBC__ */
-#error "no getprogname implementation available"
-#endif
-
 #endif /* !HAVE_GETPROGNAME */
